orchestrator/test_runner/src/erc_721_happy_path.rs

```diff
@@ -0,0 +1,638 @@
+        .to_address(&contact.get_prefix())
+        .unwrap();
+    let coin = contact
+        .get_balance(dest_cosmos_address, format!("gravity{}", erc20_contract))
+        .await
+        .unwrap()
+        .unwrap();
+    let token_name = coin.denom;
+    let amount = coin.amount;
+
+    let bridge_denom_fee = Coin {
+        denom: token_name.clone(),
+        amount: 1u64.into(),
+    };
+    let amount = amount - 5u64.into();
+    info!(
+        "Sending {}{} from {} on Cosmos back to Ethereum",
+        amount, token_name, dest_cosmos_address
+    );
+
+    let res = send_to_eth(
+        dest_cosmos_private_key,
+        dest_eth_address,
+        Coin {
+            denom: token_name.clone(),
+            amount: amount.clone(),
+        },
+        bridge_denom_fee.clone(),
+        bridge_denom_fee.clone(),
+        contact,
+    )
+    .await
+    .unwrap();
+    info!("Sent tokens to Ethereum with {:?}", res);
+
+    contact.wait_for_next_block(TOTAL_TIMEOUT).await.unwrap();
+    let requester_address = requester_cosmos_private_key
+        .to_address(&contact.get_prefix())
+        .unwrap();
+    get_oldest_unsigned_transaction_batches(
+        &mut grpc_client,
+        requester_address,
+        contact.get_prefix(),
+    )
+    .await
+    .expect("Failed to get batch to sign");
+
+    let mut current_eth_batch_nonce =
+        get_tx_batch_nonce(gravity_address, erc20_contract, *MINER_ADDRESS, web30)
+            .await
+            .expect("Failed to get current eth valset");
+    let starting_batch_nonce = current_eth_batch_nonce;
+
+    let start = Instant::now();
+    while starting_batch_nonce == current_eth_batch_nonce {
+        info!(
+            "Batch is not yet submitted {}>, waiting",
+            starting_batch_nonce
+        );
+        current_eth_batch_nonce =
+            get_tx_batch_nonce(gravity_address, erc20_contract, *MINER_ADDRESS, web30)
+                .await
+                .expect("Failed to get current eth tx batch nonce");
+        delay_for(Duration::from_secs(4)).await;
+        if Instant::now() - start > TOTAL_TIMEOUT {
+            panic!("Failed to submit transaction batch set");
+        }
+    }
+
+    if web30.eth_get_balance(dest_eth_address).await.unwrap() == 0u8.into() {
+        // we have to send this address one eth so that it can perform contract calls
+        send_one_eth(dest_eth_address, web30).await;
+    }
+
+    check_erc20_balance(erc20_contract, amount.clone(), dest_eth_address, web30).await;
+    info!(
+        "Successfully updated txbatch nonce to {} and sent {}{} tokens to Ethereum!",
+        current_eth_batch_nonce, amount, token_name
+    );
+}
+
+// this function submits a EthereumBridgeDepositClaim to the module with a given nonce. This can be set to be a nonce that has
+// already been submitted to test the nonce functionality.
+#[allow(clippy::too_many_arguments)]
+async fn submit_duplicate_erc20_send(
```

Comment:
this needs to be updated for erc721

Reply:
Will add this back in after code review of now working functionality